app/containers/HomeContainer/index.js

```diff
@@ -37,89 +46,112 @@ const Container = styled.div`
+    const loaded = get(itunesData, 'results', null) || itunesError;
     if (loading && loaded) {
       setLoading(false);
     }
-  }, [reposData]);
+  }, [itunesData]);
 
   useEffect(() => {
-    if (repoName && !reposData?.items?.length) {
-      dispatchGithubRepos(repoName);
+    if (artistName && !itunesData) {
+      dispatchSongs(artistName);
+      console.log(dispatchSongs('abcd'));
       setLoading(true);
     }
   }, []);
-
   const history = useHistory();
 
   const handleOnChange = rName => {
     if (!isEmpty(rName)) {
-      dispatchGithubRepos(rName);
+      dispatchSongs(rName);
       setLoading(true);
     } else {
-      dispatchClearGithubRepos();
+      dispatchClearSongs();
     }
   };
   const debouncedHandleOnChange = debounce(handleOnChange, 200);
 
-  const renderRepoList = () => {
-    const items = get(reposData, 'items', []);
-    const totalCount = get(reposData, 'totalCount', 0);
+  const playAudio = (url, trackId) => {
+    
+    setIsPlaying(true);
+    setTuneId(trackId);
+    audio?.pause();
+    let curAudio = new Audio(url);
+    setAudio(curAudio);
+   if(curTime>0 && url===trackUrl){
+     curAudio.currentTime =curTime
+   }
+    setTrackUrl(url)
+    curAudio.play()
+    
+  };
+  const pauseAudio = (url) => {
+    setIsPlaying(false);
+    audio?.pause();
+    setCurTime(audio?.currentTime)
+  };
+  const RenderRepoList = () => {
+    const items = get(itunesData, 'results', []);
+    const resultCount = get(itunesData, 'resultCount', 0);
     return (
       (items.length !== 0 || loading) && (
         <CustomCard>
           <Skeleton loading={loading} active>
-            {repoName && (
+            {artistName && (
               <div>
-                <T id="search_query" values={{ repoName }} />
+                <T id="search_query" values={{ artistName }} />
               </div>
             )}
-            {totalCount !== 0 && (
+            {resultCount !== 0 && (
               <div>
-                <T id="matching_repos" values={{ totalCount }} />
+                <T id="matching_repos" values={{ resultCount }} />
               </div>
             )}
-            {items.map((item, index) => (
-              <CustomCard key={index}>
-                <T id="repository_name" values={{ name: item.name }} />
-                <T id="repository_full_name" values={{ fullName: item.fullName }} />
-                <T id="repository_stars" values={{ stars: item.stargazersCount }} />
-              </CustomCard>
-            ))}
+            <Row gutter={{ xs: 8, sm: 16, md: 24, lg: 32 }}>
+              {items.map((item, index) => {
+                return (
+                  <Col key={index} className="gutter-row" span={6}>
+                    <TrackCard item={item} condition={isPlaying && item.trackId === tuneId} value={item.trackId === tuneId && audio.currentTime!==0 ?curTime:0} max={`${item.trackTimeMillis/10000}`} onPlay={playAudio} onPause={pauseAudio}></TrackCard>
```

Comment:
If a component doesn't have children you don't need to add another closing tag, you can do something like this

```
<TrackCard {...props} />
```
TrackCard should only accept track as a prop the rest of the stuff should be calculated inside

<TrackCard 
  track={item} 
  currentlyPlayingTrackId={trackId}
  currentTime={audio?.currentTime}
  onPlay={playAudio} 
  onPause={pauseAudio} />